perun-core/src/main/java/cz/metacentrum/perun/core/entry/ConsentsManagerEntry.java

```diff
@@ -139,4 +145,27 @@ public ConsentHub updateConsentHub(PerunSession sess, ConsentHub consentHub) thr
 
 		return getConsentsManagerBl().updateConsentHub(sess, consentHub);
 	}
+	@Override
+	public Consent changeConsentStatus(PerunSession sess, int consentId, ConsentStatus status, int userId) throws ConsentNotExistsException, PrivilegeException, UserNotExistsException, InvalidConsentStatus {
+		Utils.notNull(sess, "sess");
+
+		Consent consent = getPerunBl().getConsentsManagerBl().getConsentById(sess, consentId);
+		User user = getPerunBl().getUsersManager().getUserById(sess, userId);
+
+		if (status != ConsentStatus.GRANTED && status != ConsentStatus.REVOKED) {
+			throw new InvalidConsentStatus("Invalid consent status value.");
+		}
+		if (status == consent.getStatus()) {
+			throw new InvalidConsentStatus("Tried to set consent status on current value.");
+		}
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "changeConsentStatus_int_ConsentStatus_int_policy", user)) {
```

Comment:
we should check this for the consent's user, not for the user who was sent with this request. otherwise any user would be authorized to change any consent's status. speaking of which, do we need the user argument in this method?